### DISCUSSION TITLE
Django app

### DISCUSSION THREAD

**Discussion by marcrleonard** [zauberzeug/nicegui#528] [Ideas / Feature Requests]
Great work with NiceGUI! It's a fantastic balance between high-level and low-level to make guis in python!
I also love that you can incorporate it into an existing fastapi app which would have things like authentication already built in. 

I was curious if there was any plan to create a django addon/app for this? It would require divorcing from fastapi, so the code model may have to change a bit (core vs ...), so I'm guessing it wouldn't be trivial.

**Comment by rodja** [comment]
Yes, it would not be trivial. But there are a lot of Django users out there. If someone wants to tackle it we could support with know-how transfer.

**Reply by marcrleonard** [reply]
I'm definitely curious to look into it! Where would you suggest I begin digging around first?

**Reply by rodja** [reply]
I suggest you start with [nicegui.py](https://github.com/zauberzeug/nicegui/blob/main/nicegui/nicegui.py). That's where we configure the FastAPI server. Also [run.py](https://github.com/zauberzeug/nicegui/blob/main/nicegui/run.py) may be of interest, because this is where the webserver is started.

If you go a bit deeper you will encounter [client.py](https://github.com/zauberzeug/nicegui/blob/main/nicegui/client.py) which builds the html response and [page.py](https://github.com/zauberzeug/nicegui/blob/main/nicegui/page.py) where the magic of the `@ui.page` decorator is encapsulated.

**Comment by innovus-h** [comment]
Any follow up to this?

**Comment by rodja** [comment]
There is a [Reddit post about Django 5.0 becoming more ASGI (async/await)](https://www.reddit.com/r/nicegui/comments/18qpqy6/nicegui_django/) which allows for better integration with NiceGUI/FastAPI.

**Comment by Mte90** [comment]
I discovered https://github.com/rexsum420/django-nice that requires 2 different server for the ports but it is not very good documented.

Right now I was able to get everything including getting the data from django but it is not ver easy to understand from the readme compared to the code.

**Reply by Mte90** [reply]
that package now is better with a new readme and a login example
